crates/ntx-builder/src/store.rs

```diff
@@ -106,7 +106,9 @@ impl StoreClient {
 
     /// Returns the list of unconsumed network notes.
     #[instrument(target = COMPONENT, name = "store.client.get_unconsumed_network_notes", skip_all, err)]
-    pub async fn get_unconsumed_network_notes(&self) -> Result<Vec<NetworkNote>, StoreError> {
+    pub async fn get_unconsumed_network_notes(
+        &self,
+    ) -> Result<Vec<SingleTargetNetworkNote>, StoreError> {
```

Comment:
I am not sure this is correct. `get_unconsumed_network_notes()` would return all network notes from the store and so this should also do the same. Also, currently, if there are non-single-target network notes, this will result in an error, I think.

Instead, we should keep this returning all network notes, and the caller should filter out the ones that are not single-target.